rasa_nlu_examples/classifiers/sparse_sklearn_intent_classifier.py

```diff
@@ -0,0 +1,267 @@
+        clf: "sklearn.model_selection.GridSearchCV" = None,
+        le: Optional["sklearn.preprocessing.LabelEncoder"] = None,
+    ) -> None:
+        """Construct a new intent classifier using the sklearn framework."""
+        from sklearn.preprocessing import LabelEncoder
+
+        super().__init__(component_config)
+
+        if le is not None:
+            self.le = le
+        else:
+            self.le = LabelEncoder()
+        self.clf = clf
+
+    @classmethod
+    def required_packages(cls) -> List[Text]:
+        return ["sklearn"]
+
+    def transform_labels_str2num(self, labels: List[Text]) -> np.ndarray:
+        """Transforms a list of strings into numeric label representation.
+        :param labels: List of labels to convert to numeric representation"""
+
+        return self.le.fit_transform(labels)
+
+    def transform_labels_num2str(self, y: np.ndarray) -> np.ndarray:
+        """Transforms a list of strings into numeric label representation.
+        :param y: List of labels to convert to numeric representation"""
+
+        return self.le.inverse_transform(y)
+
+    def train(
+        self,
+        training_data: TrainingData,
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Train the intent classifier on a data set."""
+
+        num_threads = kwargs.get("num_threads", 1)
+
+        X, y = self.prepare_data(training_data)
+        self.clf = self._create_classifier(num_threads, y)
+
+        with warnings.catch_warnings():
+            # sklearn raises lots of
+            # "UndefinedMetricWarning: F - score is ill - defined"
+            # if there are few intent examples, this is needed to prevent it
+            warnings.simplefilter("ignore")
+            self.clf.fit(X, y)
+
+    def prepare_data(
+        self, training_data: TrainingData
+    ) -> Tuple[scipy.sparse.spmatrix, np.ndarray]:
+        """
+        Converts a rasa TrainingData object into a tuple of a sparse feature
+        matrix and a dense vector of labels.
+        """
+
+        labels = [e.get("intent") for e in training_data.intent_examples]
+
+        if len(set(labels)) < 2:
+            rasa.shared.utils.io.raise_warning(
+                "Can not train an intent classifier as there are not "
+                "enough intents. Need at least 2 different intents. "
+                "Skipping training of intent classifier.",
+                docs=DOCS_URL_TRAINING_DATA_NLU,
+            )
+            return
+
+        y = self.transform_labels_str2num(labels)
+        X = scipy.sparse.vstack(
+            [
+                self._get_sentence_features(example)
+                for example in training_data.intent_examples
+            ]
+        )
+
+        return X, y
+
+    @staticmethod
+    def _get_sentence_features(message: Message) -> scipy.sparse.spmatrix:
+        _, sentence_features = message.get_sparse_features(TEXT)
```

Comment:
What happens if we add dense data to our pipeline too? It might be good to raise a warning.

Reply:
How does this warning message strike you, "Dense features are being computed but not used in the SparseNaiveBayesIntentClassifier." ?

Reply:
That sounds perfect :)